src/components/PromotionList.tsx

```diff
@@ -29,6 +30,33 @@ export default function PromotionList({
     ...styles.container,
   };
 
+  /**
+   * If the user has not saved the promotion, save the promotion. Otherwise, delete it from their saved promotions.
+   */
+  const onLikeButtonClick = useCallback(
+    (promotionId: string) => {
+      const promos = [...promotions];
+      const promotion = promos.find(({ id }) => id === promotionId);
+      if (promotion) {
+        if (promotion.liked) {
+          return UserService.savePromotion(promotionId)
+            .then(() => {
+              promotion.liked = false;
+              setPromotions(promos);
+            })
+            .catch(() => null);
+        }
+        return UserService.unsavePromotion(promotionId)
+          .then(() => {
+            promotion.liked = true;
+            setPromotions(promos);
+          })
+          .catch(() => null);
+      }
+    },
+    [promotions, setPromotions]
+  );
```

Comment:
How would we remember whether the promotion is `liked` if we refresh the page? Because it seems like `liked` isn't a property we have in the BE for a promotion, so it would default to false once we refresh the page right.

But it does make sense for the BE to provide the `liked` property though for a promotion...the BE would need to know which user is making request for GET promotions. Hmmm this will require some refactoring in the BE.

Reply:
I'm thinking we might have to change these two routes to include the `id` of the user. Because now these endpoints need to have context of the user. But at the same time, I'm not a huge fan of making these major endpoints user specific. I think I'll need to research different ways of doing this

https://github.com/ubclaunchpad/PromoPal-backend/blob/master/src/main/route/PromotionRouter.ts#L12-L13

Reply:
@ericrkuo What do you think about just creating a duplicate endpoint for the first one like `GET /promotions/:userId` which returns the liked property? I'm not sure if we need the second one with the promotionId

Reply:
I think if we've authorized the request, we can use the user in the request headers. Then we don't need to make a new endpoint, we just modify the existing one to use the user based on the authorization

Reply:
At the same time, I think doing this is risky in terms of we need user auth in the FE to be complete first before we make the GET promotions endpoint authorized.

Reply:
Oh yeah that's a good point, forgot about that. And I guess that makes it easy because if a user isn't logged in then we'd just fetch all the promotions without joining

I think it's ok if we do it now actually since Yu merged his work in. Running the most up-to-date versions of the FE and the BE together is currently already broken because of user auth :(

Reply:
Oh it's broken at the moment? Hmm, probably next time we shouldn't merge in the auth without having it done in the FE too.

Hmm if the user isn't logged in, I think the whole endpoint wouldn't work because our authentication would fail at the middleware level. That's why I'm kind of thinking it's very complex to have our application work for both non-authenticated (users who did not sign up) and authenticated users.

I can work on implementing the changes in GET /promotions, but I think we should fail the request if the user is not authenticated and probably if we can, abandon the idea of having our app work without users signing up first

Reply:
Actually, I think we don't need to put user auth in the GET /promotions, just an optional parameter `userId` in the request params. But I still think we should consider avoiding making our application behave differently for non-auth vs auth users.

Reply:
Oh sorry maybe it's just the Uploaded Promotions but I think it's just the User endpoints because of the new authentication, but yeah you're right we should have waited until they were both complete

And yeah those are good points, it does also increase complexity on the FE to have to provide an unauthenticated view on top of the authenticated one. I think in terms of UX it probably will deter usage to make them sign up but for meeting our deadline it probably isn't the largest concern? So yeah for now let's abandon that idea

Reply:
Okay yeah that makes sense!

Reply:
Okie I think I can get this done by Wed night, would that be okay?

Reply:
Oh yes that would be great! No rush if you're busy but let me know when you get to it :)